Detectors/PHOS/calib/PHOSCalibWorkflow/include/PHOSCalibWorkflow/RingBuffer.h

```diff
@@ -0,0 +1,81 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_CALIBRATION_PHOS_RINGBUFFER_H
+#define O2_CALIBRATION_PHOS_RINGBUFFER_H
+
+/// @file   RingBuffer.h
+/// @brief  Device to collect energy and time PHOS energy and time calibration.
+
+#include <TLorentzVector.h>
+#include <TVector3.h>
+#include <array>
+
+namespace o2
+{
+namespace phos
+{
+
+// For real/mixed distribution calculation
+class RingBuffer
+{
+ public:
+  RingBuffer() = default;
+  ~RingBuffer() = default;
+
+  short size()
+  {
+    if (mFilled) {
+      return kBufferSize;
+    } else {
+      return mCurrent;
+    }
+  }
+  void addEntry(TLorentzVector& v)
+  {
+    mBuffer[mCurrent] = v;
+    mCurrent++;
+    if (mCurrent >= kBufferSize) {
+      mFilled = true;
+      mCurrent -= kBufferSize;
+    }
+  }
+  const TLorentzVector getEntry(short index)
```

Comment:
should not it be `const TLorentzVector& getEntry(short index) const` ? Otherwise I don't see the reason for `const`

Reply:
fixed